clients/include/testing_gemm_ex.hpp

```diff
@@ -241,6 +241,61 @@ void testing_gemm_ex_bad_arg(const Arguments& arg)
                           rocblas_status_invalid_handle);
 }
 
+template <typename Ti, typename To, typename Tc>
+void reference_gemm(rocblas_operation transA,
+                    rocblas_operation transB,
+                    rocblas_int       m,
+                    rocblas_int       n,
+                    rocblas_int       k,
+                    Tc                alpha,
+                    Ti*               A,
+                    rocblas_int       lda,
+                    Ti*               B,
+                    rocblas_int       ldb,
+                    Tc                beta,
+                    To*               C,
+                    rocblas_int       ldc)
+{
+    cblas_gemm<Ti, To, Tc>(transA, transB, m, n, k, alpha, A, lda, B, ldb, beta, C, ldc);
+}
+
+namespace
+{
+    bool is_replacement_kernel(rocblas_int m, rocblas_int n)
+    {
+        if((m == 512 && n == 512) || (m == 1024 && n == 1024) || (m == 2048 && n == 2048)
+           || (m == 4096 && n == 4096) || (m == 960 && n == 1024) || (m == 3840 && n == 4096))
+            return true;
+        return false;
```

Comment:
Why are certain sizes singled out as far as expected answers?

Our tests should usually be "black-box", i.e., without knowledge of internal implementation details. So we should not single out certain sizes as behaving differently, but we should apply the same standard to all sizes.

For some precisions, this might require the use of `near` instead of `unit` testing functions.

Reply:
Those particular sizes are currently implemented using replacement kernels that we want to test. The replacement kernel truncates the answer rather than rounds so a different test behaviour is needed. The test is already using a near check rather than a unit check.